### PR TITLE
8321480: ISO 4217 Amendment 176 Update

### PR DESCRIPTION
**Pull request by Karm**
backport ea06b5c JDK11 JDK-8321480 ISO 4217 Amendment 176 Update

depends on #559

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8321480](https://bugs.openjdk.org/browse/JDK-8321480) needs maintainer approval

### Issue
 * [JDK-8321480](https://bugs.openjdk.org/browse/JDK-8321480): ISO 4217 Amendment 176 Update (**Bug** - P3 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/561/head:pull/561` \
`$ git checkout pull/561`

Update a local copy of the PR: \
`$ git checkout pull/561` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/561/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 561`

View PR using the GUI difftool: \
`$ git pr show -t 561`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/561.diff">https://git.openjdk.org/jdk8u-dev/pull/561.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/561#issuecomment-2288793080)

### CODE REVIEW

**Comment by Karm** [comment]
/approval request Backporting regular ISO 4217 update from JDK 11. It's currency data update plus bringing along a workaround for missing JDK-8310923 (no JUnit refactoring). Passes tests in jdk/test/java/util/Currency/.

**Comment by Karm** [comment]
Faild test:
```
FAILED: security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java#sslrooteccca
...
Caused by: java.io.IOException: Server returned HTTP response code: 502 for URL: http://ocsps.ssl.com/
...
```

IMHO OCSP responder is just down...

File: jdk/test/java/util/Currency/ValidateISO4217.java [hunk 1]
```diff
@@ -23,8 +23,8 @@
 /*
  * @test
  * @bug 4691089 4819436 4942982 5104960 6544471 6627549 7066203 7195759
- *      8074350 8074351 8145952 8187946 8193552 8202026 8204269
- *      8208746 8209775 8274658 8283277 8296239
+ *      8039317 8074350 8074351 8145952 8187946 8193552 8202026 8204269
+ *      8208746 8209775 8264792 8274658 8283277 8296239 8321480
```
Comment (jerboaa) [inline jdk/test/java/util/Currency/ValidateISO4217.java:27]: This should only be a one-line change appending `8321480` to the last line. See the [JDK 11u commit](https://github.com/openjdk/jdk11u-dev/commit/ea06b5c966e9261cacb35300ddde6eae48ec632f#diff-07e0488dd131c8c4cfcf7c135041095716c42fdeacaf67e8f37475bf9c4780ffR27) which does the same thing.

**Review by jerboaa — COMMENTED** [review]
What testing have you done?

**Comment by Karm** [comment]
@jerboaa 
> What testing have you done?

```
karm@laptop:~/Tools/jtreg/build/images/jtreg/bin ((jtreg-7.4+1))$
TIME=`date +%s`;mkdir -p test.${TIME}/jdk/JTwork test.${TIME}/jdk/JTreport; jtreg -a -ignore:quiet 
-w:test.${TIME}/jdk/JTwork -r:test.${TIME}/jdk/JTreport -jdk:/home/karm/workspaceRH/jdk8u/build/linux-x86_64-normal-server-release/images/j2sdk-image/ 
/home/karm/workspaceRH/jdk8u/jdk/test/java/util/Currency/

Test results: passed: 5
Report written to /home/karm/Tools/jtreg/build/images/jtreg/bin/test.1724152749/jdk/JTreport/html/report.html
Results written to /home/karm/Tools/jtreg/build/images/jtreg/bin/test.1724152749/jdk/JTwork
```

File: jdk/test/java/util/Currency/ValidateISO4217.java (hunk 1 above)
Comment (Karm) [reply jdk/test/java/util/Currency/ValidateISO4217.java]: @jerboaa Ack.  bc18ee3

**Review by jerboaa — APPROVED** [review]

**Pull request closed without merging**